File: 0410-split-array-largest-sum/0410-split-array-largest-sum.py
class Solution(object):
    def isPossible(self,nums,k,mid):
        k -=1
        sum=0
        for i in nums:
            if(i>mid):
                return False
            if(sum+i>mid):
                k -=1
                sum=0
            sum +=i
        return k>=0
    def splitArray(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        min_v=min(nums)
        max_v=sum(nums)
        ans=0
        while(min_v<=max_v):
            mid=(min_v+max_v)/2
            if(self.isPossible(nums,k,mid)):
                ans=mid
                max_v=mid-1
            else:
                min_v=mid+1
        return ans
        

File: 0410-split-array-largest-sum/0410-split-array-largest-sum.java
class Solution {
    public boolean isPossible(int[] nums,int k,int mid){
        k--;
        int sum=0;
        for(int i:nums){
            if(i > mid) return false;
            if(sum+i>mid){
                k--;
                sum =0;
            }
            sum +=i;
        }
        return k>=0;

    }
    public int splitArray(int[] nums, int k) {
        
        int min=Integer.MAX_VALUE;
        int max=0;
        for(int i:nums){
            min=Math.min(min,i);
            max+=i;
        }
        int ans=0;
        while(min<=max){
            int mid=(min+max)/2;
            if(isPossible(nums,k,mid)){
                ans=mid;
                max = mid-1;
            }
            else{
                min=mid+1;
            }
        }
        return ans;
    }
}